unittests/Parse/LexerTests.cpp

```diff
@@ -731,29 +763,6 @@ TEST_F(LexerTest, DiagnoseEmbeddedNul) {
                              "1, 4: nul character embedded in middle of file"));
 }
 
-TEST_F(LexerTest, DiagnoseEmbeddedNulOffset) {
-  const char Source[] = " \0 \0 aaa \0 \0 bbb";
-  size_t SourceLen = sizeof(Source) - 1;
-
-  LangOptions LangOpts;
-  SourceManager SourceMgr;
-  unsigned BufferID = SourceMgr.addMemBufferCopy(StringRef(Source, SourceLen));
-
-  StringCaptureDiagnosticConsumer DiagConsumer;
-  DiagnosticEngine Diags(SourceMgr);
-  Diags.addConsumer(DiagConsumer);
-
-  Lexer L(LangOpts, SourceMgr, BufferID, &Diags,
-          LexerMode::Swift, HashbangMode::Disallowed,
-          CommentRetentionMode::None,
-          /*Offset=*/5, /*EndOffset=*/SourceLen);
-
-  ASSERT_FALSE(containsPrefix(
```

Comment:
Do we need these two identical assertions here?

Reply:
They aren’t quite identical. The first assertion checks that we don’t get a diagnostic for the null byte at offset 2, the other one that we don’t get one for the null byte at offset 4.